### PR TITLE
Adds documentation to the Dealer and DeckParser classes

### PR DESCRIPTION
**Pull request by BelfDev**
- Organize the code inside ```Dealer```
- Add proper documentation
- Enhance English grammar

_Thanks @rjs68 for the input!_

### CODE REVIEW

File: src/main/java/com/toptrumps/core/card/Dealer.java [hunk 1]
```diff
@@ -9,38 +9,68 @@
 
 import static java.util.stream.Collectors.toCollection;
 
+/**
+ * This class is responsible for handling fundamental card collection operations.
+ * It is parses a text file into a list of cards (deck), and provides methods for
```
Comment (rjs68) [inline src/main/java/com/toptrumps/core/card/Dealer.java]: Remove "is"

File: src/main/java/com/toptrumps/core/card/DeckParser.java [hunk 2]
```diff
@@ -6,15 +6,18 @@
...
-
+/**
+ * Utility class to handle deck parsing.
+ */
 class DeckParser {
 
     private final static String ITEM_SEPARATOR = " ";
 
-    private DeckParser() {}
+    // Prevents someone from creating an DeckParser instance
```
Comment (rjs68) [inline src/main/java/com/toptrumps/core/card/DeckParser.java]: "a" instead of "an"

**Review by rjs68 — APPROVED** [review]
2 minor things but otherwise it's awesome!

File: src/main/java/com/toptrumps/core/card/Dealer.java (hunk 1 above)
Comment (BelfDev) [reply src/main/java/com/toptrumps/core/card/Dealer.java]: Outdated

File: src/main/java/com/toptrumps/core/card/DeckParser.java (hunk 2 above)
Comment (BelfDev) [reply src/main/java/com/toptrumps/core/card/DeckParser.java]: lol, sorry.

**Review by hughwin — APPROVED** [review]

**Review by rjs68 — APPROVED** [review]

**Pull request merged**